.github/workflows/test-download-artifact-s3-action.yml

```diff
@@ -0,0 +1,118 @@
+name: Test Download Artifact S3 Action
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/actions/download-artifact-s3/**'
+      - '!**/*.md'
+  pull_request:
+    paths:
+      - '.github/actions/download-artifact-s3/**'
+      - '!**/*.md'
+
+jobs:
+
+  build:
+    name: test-download-artifact-s3-build
+
+    strategy:
+      matrix:
+        runs-on: [ubuntu-latest, macos-latest, windows-latest]
+      fail-fast: false
+      max-parallel: 1
+
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set Node.js 22.x
+      uses: actions/setup-node@v4
+      with:
+        node-version: 22.x
+
+    - name: npm install
+      run: npm install
+      working-directory: '.github/actions/download-artifact-s3'
+
+    - name: Compile
+      run: npm run build
+      working-directory: '.github/actions/download-artifact-s3'
+
+    - name: Lint
+      run: npm run lint
+      working-directory: '.github/actions/download-artifact-s3'
+
+    - name: Format
+      run: npm run format-check
+      working-directory: '.github/actions/download-artifact-s3'
+
+    # Test end-to-end by uploading two artifacts and then downloading them
+    - name: Create artifacts
+      run: |
+        mkdir -p path/to/artifact-A
+        mkdir -p path/to/artifact-B
+        echo "Lorem ipsum dolor sit amet" > path/to/artifact-A/file-A.txt
+        echo "Hello world from file B" > path/to/artifact-B/file-B.txt
+
+    - name: Upload artifact A
+      uses: actions/upload-artifact@v4
+      with:
+        name: Artifact-A-${{ matrix.runs-on }}
+        path: path/to/artifact-A
+
+    - name: Upload artifact B
+      uses: actions/upload-artifact@v4
+      with:
+        name: Artifact-B-${{ matrix.runs-on }}
+        path: path/to/artifact-B
```

Comment:
Why do we need these?

Reply:
These are the uploads that were in your original test to try the migrated download action.
Uploads are done using the original "actions/upload-artifact" action into GHA artifacts and then we call the migrated "download-artifact-s3" action to download these files. (Part of the unit test)

The code is pretty much the same compared to seemethere/download-artifact-s3 with the exception that I added the unique name so that it wont conflict with each run from the matrix definition.